src/sidebar/components/button.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const classnames = require('classnames');
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+
+const SvgIcon = require('./svg-icon');
+
+/**
+ * A button, which can be one of three core types depending on provided props:
+ * - icon-only (`icon` set, `label` missing)
+ * - text-only (`label` set, `icon` missing)
+ * - icon and text (`icon` and `label` present)
+ *
+ * In addition, a button may be adjusted:
+ * - `useCompactStyle`: to fit in a tighter spot
+ * - `useInputStyle`: to sit next to an input element
+ */
+function Button({
+  className = '',
+  icon = '',
+  isActive = false,
+  isPrimary = false,
+  label,
+  onClick = () => null,
+  title,
+  useCompactStyle = false,
+  useInputStyle = false,
```

Comment:
We're using a mix of naming conventions here for the various button varieties (`isPrimary` vs `useInputStyle`). It would be ideal to have a consistent convention for such flags. Also, I wonder if we can make it clearer which of these styles are mutually exclusive and which can be combined together.